README.md

```diff
@@ -480,40 +480,156 @@ that the wired class extends, instead of the full implementation.
+//this compiles to:
+//lazy val theUserFinder = system.actorOf(Props(classOf[UserFinderActor], theDatabaseAccess, theSecurityFilter), "userFinder")
+``` 
+
+Note how easy is to create actorRef:
+
+```scala
+wireActor[UserFinderActor]("userFinder")
+```
+ 
+compared to:
+ 
+```scala
+system.actorOf(Props(classOf[UserFinderActor], theDatabaseAccess, theSecurityFilter), "userFinder")
+```
+
+In order to make it working all dependencies created `Actor`'s (`UserFinderActor`'s) primary constructor and 
+instance of the `akka.actor.ActorRefFactory` must be in scope. In above example this is all true. Dependencies
+of the `UserFinderActor` are `DatabaseAccess` and `SecurityFilter` and they are in scope. 
+The `ActorRefFactory` is in scope as well because `ActorSystem` which is subtype of it is there.
+
+
+Using Macwire for wiring Actors can be attractive not only because it safes from to much typing. 
+It is as well type safer because possible errors will be reported by scala compiler. In standard way 
+of creating actors if parameters passed to the `Props` are mistaken
+error happens during runtime. However relying on `wireActor[A]` here compilation error is raised. 
+
+Although this code is wrong it will compile and throw exception during runtime:
+```scala
+system.actorOf(
+  Props(classOf[UserFinderActor], theDatabaseAccess),  //no theSecurityFilter 
+  "userFinder"
+)
+```
+
+However below snipped won't compile:
+
+```scala
+lazy val theDatabaseAccess = wire[DatabaseAccess]
+//no theSecurityFilter
+val system = ActorSystem("actor-system") 
+lazy val theUserFinder = wireActor[UserFinder]("userFinder") 
+//Error: Cannot find a value of type: [com.softwaremill.macwire.akkasupport.Demo.SecurityFilter]
+//val theUserFinder = wireActor[UserFinderActor]("userFinder")
+```  
 
-object Example3 {
-  val a = wire[A]
-  val actorSystem = ActorSystem()
-  val myActor1: ActorRef = system.actorOf(wireProps[MyActor])  
-  // this compiles to:
-  // val myActor1: ActorRef = system.actorOf(Props(new MyActor(a))) 
+Creating actor within another actor is even simpler than in first example because we don't need to have `ActorSystem` in scope. 
+The `ActorRefFactory` is here because `Actor.context` is subtype of it. Let's see this in action:
+
+```scala
+class UserStatusReaderActor(theDatabaseAccess: DatabaseAccess) extends Actor {
+  val theSecurityFilter = wire[SecurityFilter]
+  
+  val userFinder = wireActor[UserFinderActor]("userFinder")
+  //this compiles to:
+  //val userFinder = context.actorOf(Props(classOf[UserFinderActor], theDatabaseAccess, theSecurityFilter), "userFinder")
+  
+  override def receive = ...
 }
-````
+```
+
+The difference is that previously macro expanded into `system.actorOf(...)` 
+and when inside another actor it expanded into `context.actoOf(...)`.
+  
+    
+It's possible to create anonymous actors. `wireAnonymousActor` is for it:
+
+```scala
+val userFinder = wireAnonymousActor[UserFinderActor]
```

Comment:
I've made assumption that user is responsible for knowing that `wireActor` will be expanded into `actorRefFactory.actorOf(...)`. This makes no assumptions about what is the implementation of `ActorRefFactory`. Therefore it's up to the user to remember that `wireActor` within another actor will create child actor and outside the actor (if `ActorSystem` is in scope) it will result in top level actor. 

IMHO this is much cleaner having `wireActor` and `wireAnonymousActor` than having `wire(TopLevel)Actor`, `wireChildActor`, `wireAnonymous(TopLevel)Actor` and `wireAnonymousChildActor` functions.

Is there any gain which I am not aware of for making such split?

@adamw, what do you think?